src/resource/user/user.service.ts

```diff
@@ -134,4 +134,24 @@ export class UserService {
       },
     });
   }
+
+  async updateProfile({ username, displayName, bio }: UpdateProfileInput) {
+    const updatedData = {
+      displayName,
+      bio,
+    };
+    const updatedUser = await this.prisma.user.update({
+      where: { username },
```

Comment:
Well. our user has a unique username, but It's better if we are updating the user with their id instead.

Reply:
Fixed: user profile is updated with `id` instead